src/dd/RealNumber.cpp

```diff
@@ -13,16 +13,20 @@
 #include "dd/DDDefinitions.hpp"
 
 #include <cassert>
-#include <cstddef>
 #include <cstdint>
 #include <cstdlib>
 #include <istream>
-#include <limits>
 #include <ostream>
 
 namespace dd {
 
-static constexpr std::size_t LSB = static_cast<std::uintptr_t>(1U);
+static constexpr std::uintptr_t LSB = 1U;
+static constexpr std::uintptr_t LSB2 = 2U;
+
+RealNumber* RealNumber::next() const noexcept {
+  const std::uintptr_t ptr = reinterpret_cast<std::uintptr_t>(next_) & ~LSB2;
+  return ptr == 0U ? nullptr : reinterpret_cast<RealNumber*>(ptr);
```

Comment:
Is that if really necessary? I would have assumed that casting 0 to any pointer type would yield nullptr 🧐